### PR TITLE
GOON MEDICAL TAKE TWO: Drugs, Meds, and Crit

### PR DESCRIPTION
**Pull request by Iamgoofball**
https://www.dropbox.com/s/pg77eeqq0deiody/I%20DO%20COKE.mp4?dl=0

:cl: Iamgoofball
experimental: GOON MEDICAL EARLY BETA TESTERS READ THIS:
experimental: ------------------------------------------------------------------------
experimental: This is a test. I've checked out 90% of possible situations and things seem to be working. Don't assume everything is gonna be broken 5ever and that it's 100% unfix-able. You guys assume that about a lot of things and refuse to accept new shit because you don't believe it can be fixed. tl;dr stop being dumb ok
experimental: ------------------------------------------------------------------------
experimental: BLOOD CHANGES:
tweak: Blood has grown some teeth, and will actually be a concern when dropping below safe blood levels.
tweak: Blood loss can be treated with saline-glucose solution, iron, and nutrient.
rscadd: There is now a Blood Donation Stand in the lobby of Medical for you to donate your blood in injector format. You get a reward for doing so!
experimental: ------------------------------------------------------------------------
experimental: MISC CHANGES:
tweak: Medical Doctors now start with Medical Hyposprays.
tweak: These Medical Hyposprays can only hold medical reagents, and start with a premixed critical rejuv. mix great for stabilizing critical folks.
experimental: ------------------------------------------------------------------------
experimental: CRITICAL CHANGES:
tweak: Critical is now softcrit instead of hard "you're knocked out for exactly X amount of seconds den u die".
tweak: You can now limp around in the early stages of crit.
tweak: Critical has 3 "stages", Shock, Cardiac Failure, and Cardiac Arrest, in that order. They each have unique cure methods, you can check how to treat someone suffering from this with your Medical Scanner.
tweak: If you empty enough bullets into someone they're probably not gonna be getting up and limping around for more than like 2 seconds, still.
tweak: tl;dr: crit is more interesting to treat as a doctor, and you have more of a chance of surviving as a patient, provided your doctor follows basic instructions.
experimental: ------------------------------------------------------------------------
experimental: This is the first step in a path towards 1:1 accuracy in reagents compared to their goon equivs.
rscadd: DRUGS:
tweak: Krokodil, Crank, Nicotine, Bath Salts, and Methamphetamine: Effects redone to be 1:1 with Goon. Find out how it works by experimenting!
rscadd: MEDICINE:
tweak: Leporazine: Renamed to Teporone.
tweak:  Omnizine, Synaptizine, Calomel, Penetic Acid, Sal. Acid, Salbutamol, Perfluorodecalin, Ephedrine, Haloperidol, Antihol. Epinephrine, Atropine, and Morphine: Effects redone to be 1:1 with Goon. Find out how it works by experimenting!
rscadd: Ether: Morphine without the painkiller functions!
/:cl:

### CODE REVIEW

**Comment by Supermichael777** [comment]
Triggered

Honestly the only thing worth shit from goon.

**Comment by PKPenguin321** [comment]
no not this again please

**Comment by TechnoAlchemisto** [comment]
No

File: code/modules/mob/living/carbon/human/blood.dm [hunk 1]
```diff
@@ -88,26 +90,46 @@ var/const/BLOOD_VOLUME_SURVIVE = 122
...
+				if(prob(50))
+					var/effect = pick(1,0)
+					if(effect)
+						Weaken(5)
+					else
+						Paralyse(5)
 			if(0 to BLOOD_VOLUME_SURVIVE)
-				death()
+				if(prob(75))
+					add_medical_effect(/datum/medical_effect/shock)
```
Comment (Razharas) [inline code/modules/mob/living/carbon/human/blood.dm]: >0 blood
>75% chance of just shock

Near zero blood should be a cardiac arrest/something equally horrible

Comment (PKPenguin321) [inline code/modules/mob/living/carbon/human/blood.dm]: Actually no blood should straight up kill you unless you're being cryogenically frozen or some shit

File: code/modules/mob/living/carbon/human/blood.dm [hunk 2]
```diff
@@ -88,26 +90,46 @@ var/const/BLOOD_VOLUME_SURVIVE = 122
...
+					else
+						Paralyse(4)
 			if(BLOOD_VOLUME_SURVIVE to BLOOD_VOLUME_BAD)
-				oxyloss += 5
-				if(prob(15))
-					Paralyse(rand(1,3))
-					var/word = pick("dizzy","woozy","faint")
-					src << "<span class='warning'>You feel extremely [word].</span>"
+				if(prob(50))
+					add_medical_effect(/datum/medical_effect/shock)
```
Comment (Razharas) [inline code/modules/mob/living/carbon/human/blood.dm]: Cardiac failure at least

File: code/modules/mob/living/carbon/human/blood.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/blood.dm]: this does rapidly kill you though

File: code/modules/mob/living/carbon/human/blood.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/blood.dm]: 1:1 parity

File: code/modules/mob/living/carbon/human/blood.dm (hunk 1 above)
Comment (Razharas) [inline code/modules/mob/living/carbon/human/blood.dm]: Concidering death comes from too much brain damage i dont think that 0 blood should instantly kill, but effect should be absolutely horrible

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/blood.dm]: Keep in mind shock can still rapidly lead to the other 2 stages of crit and will probably kill you

**Comment by Firecage** [comment]
Again!? WHY!

**Comment by Core0verload** [comment]
+ "Universally Disliked" and "MEME PR" tags.

**Comment by Dorsisdwarf** [comment]
I'm pretty sure remaking denied PRs is against a rule somewhere

**Comment by Razharas** [comment]
Blood changes +, id wish bleeding was health effect like cardiac shit though
Misc changes -, i dont see the reason for this
Crit changes +, i like non-binary crit
Drug changes -, i dont see the reason for this

**Comment by Iamgoofball** [comment]
@Dorsisdwarf there was some misunderstanding over a meme

@Razharas misc changes are to help with treating the new medical effects and improve medbay gameolay

the drug changes are to cope with the new medical system

**Comment by mikecari** [comment]
Goof getting deported to /vg/ when?

**Comment by MMMiracles** [comment]
Never. 

Goof for president 2020

**Comment by AlexanderUlanH** [comment]
Have you ported people being unable to speak with losebreath? If not, gooncrit will absolutely wreck game balance.

File: code/modules/mob/living/carbon/human/blood.dm (hunk 1 above)
Comment (AdamElTablawy) [inline code/modules/mob/living/carbon/human/blood.dm]: shock takes a while to crit you know

having no blood should immediately induce the worst state of crit

**Comment by Iamgoofball** [comment]
Already done.

File: code/modules/mob/living/carbon/human/blood.dm [hunk 3]
```diff
@@ -88,26 +90,46 @@ var/const/BLOOD_VOLUME_SURVIVE = 122
 				if(!pale)
 					pale = 1
 					update_body()
-					var/word = pick("dizzy","woozy","faint")
-					src << "<span class='warning'>You feel [word].</span>"
-				adjustOxyLoss((BLOOD_VOLUME_NORMAL - blood_volume) / 100)
+				if(prob(15))
+					add_medical_effect(/datum/medical_effect/shock)
```
Comment (lordpidey) [inline code/modules/mob/living/carbon/human/blood.dm]: Wait what?  If I lose ONE unit of blood, there's a 15% chance I go into shock every second?

File: code/modules/mob/living/carbon/human/blood.dm [hunk 4]
```diff
@@ -70,13 +70,15 @@ var/const/BLOOD_VOLUME_SURVIVE = 122
...
 							break
 
-				B.volume += 0.1 // regenerate blood VERY slowly
 				if(reagents.has_reagent("nutriment"))	//Getting food speeds it up
 					B.volume += 0.4
 					reagents.remove_reagent("nutriment", 0.1)
 				if(reagents.has_reagent("iron"))	//Hematogen candy anyone?
 					B.volume += 0.4
 					reagents.remove_reagent("iron", 0.1)
+				if(reagents.has_reagent("salglu_solution"))	//Hematogen candy anyone?
```
Comment (lordpidey) [inline code/modules/mob/living/carbon/human/blood.dm]: Nice comment.

**Comment by lordpidey** [comment]
You're gonna want to remove, or at the very least modify saline glucose's reaction_mob, as it already increases blood amount on inject.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 5]
```diff
@@ -384,10 +436,14 @@
...
 		if(R != src)
-			M.reagents.remove_reagent(R.id,2.5)
+			M.reagents.remove_reagent(R.id,5)
 	if(M.health > 20)
 		M.adjustToxLoss(2.5*REM, 0)
 		. = 1
+	if(prob(6))
+		M.visible_message("<span class = 'danger'>[M] pukes all over \himself.</span>")
+		playsound(M.loc, "sound/effects/splat.ogg", 50, 1)
+		new /obj/effect/decal/cleanable/vomit(M.loc)
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: call
M.vomit()
instead of all of those lines.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 6]
```diff
@@ -413,13 +469,17 @@
 
 /datum/reagent/medicine/pen_acid/on_mob_life(mob/living/M)
 	if(M.radiation > 0)
-		M.radiation -= 4
-	M.adjustToxLoss(-2*REM, 0)
+		M.radiation -= 7
+	if(prob(75))
+		M.adjustToxLoss(-4*REM, 0)
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Why not just use M.adjustToxLoss(-3*REM, 0) instead of the if statement? Using probabilities are helpful to ensure health is a binary fraction, but that's not applicable here.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 7]
```diff
@@ -486,53 +538,52 @@
...
-	if(prob(33))
-		M.adjustToxLoss(5*REM, 0)
-		M.losebreath += 5
-		. = 1
+/datum/reagent/medicine/ephedrine/overdose_process(mob/living/M, severity)
+	var/effect = ..(M, severity)
+	switch(severity)
+		if(1)
+			if(effect <= 1)
+				M.visible_message("<span class = 'danger'><b>[M.name]</b> suddenly and violently vomits!</span>")
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: M.vomit()

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 8]
```diff
@@ -486,53 +538,52 @@
...
+				playsound(M.loc, "sound/effects/splat.ogg", 50, 1)
+				new /obj/effect/decal/cleanable/vomit(M.loc)
+			else if (effect <= 3)
+				M.emote(pick("groan","moan"))
+			if(effect <= 8)
+				M.adjustToxLoss(1)
+		if(2)
+			if(effect <= 2)
+				M.visible_message("<span class = 'danger'><b>[M.name]</b> suddenly and violently vomits!</span>")
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: M.vomit()

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 7 above)
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: M.vomit()

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 8 above)
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: M.vomit()

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 9]
```diff
@@ -685,33 +713,65 @@
...
+			if(C.has_medical_effect(/datum/medical_effect/heartfailure))
+				C.remove_medical_effect(/datum/medical_effect/heartfailure)
+	..()
+
+/datum/reagent/medicine/epinephrine/overdose_process(mob/living/M, severity)
+	var/effect = ..(M, severity)
+	switch(severity)
+		if(1)
+			if(effect <= 1)
+				M.visible_message("<span class = 'danger'><b>[M.name]</b> suddenly and violently vomits!</span>")
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: M.vomit()

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 10]
```diff
@@ -685,33 +713,65 @@
...
+				playsound(M.loc, "sound/effects/splat.ogg", 50, 1)
+				new /obj/effect/decal/cleanable/vomit(M.loc)
+			else if (effect <= 3)
+				M.emote(pick("groan","moan"))
+			if(effect <= 8)
+				M.adjustToxLoss(1)
+		if(2)
+			if(effect <= 2)
+				M.visible_message("<span class = 'danger'><b>[M.name]</b> suddenly and violently vomits!</span>")
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: M.vomit()

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 6 above)
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: nope, I want it to be on a prob()

Comment (PKPenguin321) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: bad goof bad

**Comment by PKPenguin321** [comment]
i pray that some day we will see a bunch of new chems without this stupid crit system tacked on

**Comment by Alek2ander** [comment]
1:1 accuracy > actually making sense

**Comment by Supermichael777** [comment]
its beter then 5 hits then y fall over 5ever

**Comment by KorPhaeron** [comment]
goof was too proud to kill the pr himself

**Comment by Iamgoofball** [comment]
fuck you the PR IS STILL ALIVE

**Comment by KorPhaeron** [comment]
I thought you asked me to close it

**Comment by Iamgoofball** [comment]
when the fuck did I do that

**Comment by TechnoAlchemisto** [comment]
kor saved us from the menace that is goofball

**Comment by Miauw62** [comment]
>1:1 with goon

goon is like some sort of code idealist absolutely supporting the transformation of our server into a goonserver by overthrowing the code bourgeoisie, followed by a short period of dictatorship by the code proletariat and then a perfect datumless society.

**Comment by GunHog** [comment]
I love how Goof has a personal tag AND milestone! What infamy! Also, is there a dummy's guide to this stuff? If this actually gets merged, I need to know what is different.

**Comment by Miauw62** [comment]
like im not even kidding goof is to goon like the ussr is to marxism

**Comment by KorPhaeron** [comment]
goof you really need to test these things a bit more before you put them on server now everyone hates it before we got to try a working version
